### ISSUE TITLE
JSpecify conformance test failures

### ISSUE THREAD

**Issue by wmdietl** [eisop/checker-framework#860]
After merging #859, we see that the unexpected error is:

````
OOPS: Basic.java:47:test:cannot-convert:capture#159 of ? extends String* to Object*
````

https://github.com/jspecify/jspecify/blob/e7f5e96c2d22a54fb8434c65db8e85b728619461/conformance-tests/src/assertions/java/org/jspecify/conformance/tests/Basic.java#L47 does not seem to pick up the JDK annotation from https://github.com/jspecify/jdk/blob/af472340eb5b8b6c249c354e81c1c84e5dbb9d62/src/java.base/share/classes/java/util/List.java#L143

So something about cloning the JDK in #821 must behave differently between a PR and a build on `master`...

**Comment by wmdietl** [comment]
Looking through the build failure:

````
++ /home/runner/work/checker-framework/checker-framework/checker/bin-devel/.git-scripts/git-clone-related eisop jdk
In directory /home/runner/work/checker-framework/checker-framework :
Cloning into '.plume-scripts'...
About to run: git clone -b master -q --single-branch --depth 1 https://github.com/eisop/jdk.git ../jdk
git-clone-related: ../jdk is at 6fe3e7815516f0e935b03f45af60f3f74ec50175
....
"$GIT_SCRIPTS/git-clone-related" jspecify jdk
+ /home/runner/work/checker-framework/checker-framework/checker/bin-devel/.git-scripts/git-clone-related jspecify jdk
In directory /home/runner/work/checker-framework/checker-framework :
About to run: git clone -b master -q --single-branch --depth 1 https://github.com/eisop/jdk.git ../jdk
git-clone-related: ../jdk is at 6fe3e7815516f0e935b03f45af60f3f74ec50175
````

In both cases the eisop/jdk is cloned!
